langtest/datahandler/datasource.py

```diff
@@ -1088,3 +1115,50 @@ def _row_to_sample_classification(self, data_row: Dict[str, str]) -> Sample:
             original=original,
             expected_results=SequenceClassificationOutput(predictions=[label]),
         )
+
+    def _row_to_ner_sample(self, data_row: dict) -> Sample:
+        """Convert a row from the dataset into a Sample for NER.
+
+        Args:
+            data_row (Dict[str, str]):
+                Single row of the dataset.
+
+        Returns:
+            Sample:
+                Row formatted into a Sample object.
+        """
+        input_column = next(
+            (col for col in self.COLUMN_NAMES["ner"]["text"] if col in data_row),
+            None,
+        )
+        output_column = next(
+            (col for col in self.COLUMN_NAMES["ner"]["ner"] if col in data_row),
+            None,
+        )
+
+        tokens = data_row.get(input_column, [])
+        labels = data_row.get(output_column, [])
+
+        #  get token and labels from the split
+        ner_labels = []
+        cursor = 0
+        for token, label in zip(tokens, labels):
+            ner_labels.append(
+                NERPrediction.from_span(
+                    entity=label,
+                    word=token,
+                    start=cursor,
+                    end=cursor + len(token),
+                    doc_id=0,
+                    doc_name="",
+                    pos_tag="XX",
+                    chunk_tag="XX",
+                )
```

Comment:
Just a random thought: we should probably get rid of `pos_tag` and `chunk_tag`, we not using them anywhere

Reply:
But not relevant in this PR